Simplify Cortex README.md

irfanpena opened the pull request:
## Describe Your Changes

- Draft the Platform readme

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

irfanpena committed:
Draft the Platform readme

irfanpena committed:
Update the Model library table

irfanpena committed:
nits

irfanpena committed:
nits

irfanpena committed:
Added the installation based on discord

irfanpena committed:
nits

irfanpena committed:
cortex-engine->cortex.cpp

irfanpena committed:
use the current banner instead

irfanpena commented:
https://github.com/janhq/cortex.cpp/issues/1036

irfanpena committed:
Merge branch 'dev' into platform-readme

dan-menlo commented on README.md:
````diff
@@ -22,22 +22,22 @@ Cortex supports the following engines:
 ## Installation
 ### MacOs
 ```bash
-brew install cortex-engine
````
This has been superseded by Installer:
https://github.com/janhq/cortex.cpp/discussions/1114

dan-menlo commented on README.md:
```diff
@@ -22,22 +22,22 @@ Cortex supports the following engines:
 ### Libraries
-- [cortex.js](https://github.com/janhq/cortex.js)
-- [cortex.py](https://github.com/janhq/cortex-python)
+- [cortex.cpp.js](https://github.com/janhq/cortex.js)
```
This will still be `cortex.js` and `cortex.py`
https://github.com/janhq/cortex.cpp/issues/1162

irfanpena replied:
Fixed

irfanpena committed:
Simplify and update the cortex.cpp readme

irfanpena committed:
Merge branch 'platform-readme' of https://github.com/janhq/cortex.cpp into platform-readme

dan-menlo commented on README.md:
```diff
@@ -5,32 +5,31 @@
 </p>
 
-> ⚠️ **Cortex is currently in Development**: Expect breaking changes and bugs!
+> ⚠️ **CortexCPP is currently in Development. This documentation outlines the intended behavior of Cortex Platform, which may not yet be fully implemented in the codebase.**
```
- We should standardize on "cortex.cpp"
- "Cortex Platform" is deprecated. We can just refer to it as "Cortex" or "cortex.cpp" from now on

dan-menlo commented on README.md:
```diff
@@ -5,32 +5,31 @@
 
 ## About
-Cortex is a C++ AI engine that comes with a Docker-like command-line interface and client libraries. It supports running AI models using `ONNX`, `TensorRT-LLM`, and `llama.cpp` engines. Cortex can function as a standalone server or be integrated as a library.
+Cortex is a C++ AI engine that comes with a Docker-like command-line interface and client libraries. Cortex can function as a standalone server or be integrated as a library.
```
I've started to standardize on this description: 

```
Cortex is a Local AI engine that is used to run and customize LLMs. Cortex can be deployed as a standalone server, or integrated into apps like Jan.ai. 
```

irfanpena replied:
The name for Cortex.cpp is Cortex or CortexCPP?

irfanpena replied:
Updated it to CortexCPP

dan-menlo commented on README.md:
````diff
@@ -5,32 +5,31 @@
 ### MacOs
 ```bash
-brew install cortex-engine
+brew install cortex.cpp
````
We no longer have `brew` - I believe we will need to download the Mac Installer, which will be a `.dmg` file. 
Please check with @hiento09

irfanpena replied:
Installation section made it into a list and link it to each respective installer link (which is empty link for now)

dan-menlo commented on README.md:
````diff
@@ -5,32 +5,31 @@
 ### Windows
 ```bash
-winget install cortex-engine
+winget install cortex.cpp
````
This will be a Windows installer, which will be a `.exe`

irfanpena replied:
For the installer. Do the users need to download from the GitHub releases? Or can they also install it using a package like Winget sudo apt?

irfanpena replied:
Installation section made it into a list and link it to each respective installer link (which is empty link for now)

dan-menlo commented on README.md:
````diff
@@ -5,32 +5,31 @@
 ### Linux
 ```bash
-sudo apt install cortex-engine
+sudo apt install cortex.cpp
````
This will be either: 

- `.deb` installer (for Debian)
- `.appimage` installer (for Fedora)

You can see the relevant discussion here: https://github.com/janhq/cortex.cpp/discussions/1114

cc @hiento09

irfanpena replied:
Installation section made it into a list and link it to each respective installer link (which is empty link for now)

dan-menlo commented on README.md:
```diff
@@ -123,73 +119,30 @@ Here are example of models that you can use based on each supported engine:
+| **Uninnstall an Engine**              | `cortex engines uninstall <engine_name> [options]`                 |
+| **Show Model Information**         | `cortex ps`                                                         |
+| **Update Cortex**         | `cortex update [options]`                                                         |
+
```
Thanks! This is a lot cleaner and more concise than the previous iteration

dan-menlo commented on platform/README.md:
```diff
@@ -4,82 +4,46 @@
 </p>
```
This entire file can be deleted, as Cortex Platform is deprecated

irfanpena replied:
File deleted

dan-menlo commented on platform/README.md:
````diff
@@ -98,42 +62,153 @@ chmod +x '[path-to]/cortex/cortex-js/dist/src/command.js'
+# Start a model
+cortex run [model_id]
+```
+## Model Library
````
Quick check: will Model Library be in the `cortex.cpp` README? (we should have it there)

We should refer to this as the Built-in Model Library

irfanpena replied:
Yes

irfanpena committed:
Update the CortexCPP readme

irfanpena committed:
Update Overview and Remove Cortex Platform readme

irfanpena committed:
nits

irfanpena committed:
Merge branch 'dev' into platform-readme

irfanpena committed:
Separate installer for each version

irfanpena committed:
Merge branch 'platform-readme' of https://github.com/janhq/cortex.cpp into platform-readme

irfanpena committed:
Update the build from source steps

irfanpena committed:
nits

irfanpena committed:
nits

irfanpena committed:
1337 -> 3928

irfanpena commented:
> @irfanpena would you mind resolving the comments you fixed so that we can lgtm? 🙏

@0xSage I have resolved the comment that I fixed; the rest I have updated, but I need your review, for example:

1. I am not really sure whether to use Cortex or CortexCPP- > based on this: https://github.com/janhq/cortex.cpp/discussions/1115#discussioncomment-10558476 the finalized name is CortexCPP, so I used CortexCPP in the readme.

2. Also for the installation section, as now we use native installers, I have updated the readme to redirect to the download links for each respective installer based on the different operating systems:

## Installation
To install CortexCPP, download the installer for your operating system from the following options:
- **Stable Version**
  - [Windows]()
  - [Mac]()
  - [Linux (Debian)]()
  - [Linux (Fedora)]()
- **Beta Version**
  - [Windows]()
  - [Mac]()
  - [Linux (Debian)]()
  - [Linux (Fedora)]()
- **Nightly Version**
  - [Windows]()
  - [Mac]()
  - [Linux (Debian)]()
  - [Linux (Fedora)]()

irfanpena committed:
Merge branch 'dev' into platform-readme

irfanpena committed:
CortexCPP -> Cortex.cpp

irfanpena committed:
Merge branch 'platform-readme' of https://github.com/janhq/cortex.cpp into platform-readme

dan-menlo commented on README.md:
```diff
@@ -1,39 +1,41 @@
-# Cortex
+# Cortex.cpp
```
Let's standardize on `Cortex.cpp`

dan-menlo commented on README.md:
````diff
@@ -1,39 +1,41 @@
-```
-### Docker
-**Coming Soon!**
+To install Cortex, download the installer for your operating system from the following options:
````
Might make sense for this to be a table, similar to Jan
https://github.com/janhq/jan?tab=readme-ov-file#download

irfanpena replied:
Updated, but still need to change the links once cortex.cpp installers published

dan-menlo commented on README.md:
```diff
@@ -43,26 +45,23 @@ sudo apt install cortex-engine
 
 To install Cortex from the source, follow the steps below:
```
Can you check with @vansangpfiev - I think there are different instructions for Mac, Windows and Linux
May make sense to push this to the bottom of the README (but link from the top)

irfanpena replied:
Updated for each operating system

dan-menlo commented on README.md:
```diff
@@ -1,39 +1,41 @@
+Cortex.cpp is a Local AI engine that is used to run and customize LLMs. Cortex can be deployed as a standalone server, or integrated into apps like [Jan.ai](https://jan.ai/).
 
-## Cortex Engines
 Cortex supports the following engines:
```
Cortex.cpp is multi-engine, but defaults to llama.cpp:

irfanpena replied:
Updated

dan-menlo commented on README.md:
```diff
@@ -122,87 +118,44 @@ Here are example of models that you can use based on each supported engine:
+
 ## REST API
-Cortex has a REST API that runs at `localhost:1337`.
+Cortex.cpp has a REST API that runs at `localhost:3928`.
```
Check with @vansangpfiev @namchuai: There is a chance we will change the port number to something less collision-prone (i.e. more digits?)

namchuai replied:
Let's bring this into discussion. I'm ok with any port number decision.

dan-menlo commented on README.md:
```diff
@@ -1,39 +1,41 @@
 </p>
 
 <p align="center">
   <a href="https://cortex.so/docs/">Documentation</a> - <a href="https://cortex.so/api-reference">API Reference</a> 
```
Can we add the links to Discord and activity links, similar to Jan? 
<img width="693" alt="image" src="https://github.com/user-attachments/assets/db45482a-4975-44ca-831d-6a8459f69f07">

irfanpena replied:
Updated

irfanpena committed:
nits

irfanpena committed:
change the engine names

dan-menlo commented:
@irfanpena I've left a few requests for change - otherwise, thank you for drafting this

irfanpena committed:
Updated per feedbacks, except the PORT

dan-menlo reviewed (approved):
@irfanpena Thanks - you can go ahead to merge this first
